### ISSUE TITLE
science mode should support an easy button to change result format from decimal mode to fractional mode

### ISSUE THREAD

**Issue by whutddk** [Microsoft/calculator#204]
**Problem Statement**

**Evidence or User Insights**

**Proposal**

science mode should support an easy button to change result format from decimal mode to fractional mode

**Goals**

when I calculate 1 divide 3, the result 0.333333333... will be given out.  But sometimes I need a more accurate result 1/3 in  fractional format. 

**Non-Goals**

**Low-Fidelity Concept**

In most of scientific calculator, a single button is added to support this funciton to satify different requirement. Can you add this in it?

**Comment by MicrosoftIssueBot** [comment]
This is your friendly Microsoft Issue Bot. I've seen this issue come in and have gone to tell a human about it.

**Comment by grochocki** [comment]
Thanks for the feature suggestion! I think this is a great start, but it seems like there are some missing details. Can you add more information to your pitch? Check out our [New Feedback Process](https://github.com/Microsoft/calculator/blob/master/docs/NewFeatureProcess.md) for more info.

This feels pretty similar to #170? Would you agree? If so, perhaps it would be better to join the discussion there.

**Comment by whutddk** [comment]
No, I think it different from #170. I think I should make some example to show, hold on.

**Comment by whutddk** [comment]
can you understand what I say? @grochocki

**Comment by whutddk** [comment]
![Uploading 1.png…]()

**Comment by Luuk34** [comment]
What if the output is: 10,00048602673147023086269744836
Should this button really change it to: 123456/12345 ?

**Comment by whutddk** [comment]
@Luuk34 Hi, the feature I metion depend on ```CASIO fx-82ES calculator```. Then I try it in real calculator,  it cannot! But I think it should,because 10,00048602673147023086269744836 = 1000048602673147023086269744836 / 10000000000000000000000000000000 ,then it can be simplified.
This function mainly use for the result like 1/3, 1/7,1/11,that can only show accurate result 
in fractional format.

**Comment by grochocki** [comment]
Thanks for the clarification @whutddk. It seems this #170 is focused on making fractions/exponents visually distinct, while your idea covers covnersion between fractions and decimals. Let's keep this idea open for discussion so the community has the chance to provide feedback.

**Comment by lzl1918** [comment]
so, it is a good idea to add similar accuracy improvements to root numbers?

Sqrt (2)=(visualized sqrt)(2)
sqrt (2.5) = sqrt (30/4) = (visualized sqrt)(30)/2
sqrt (3)/sqrt (2) = (visualized sqrt)(6)/2

**Comment by grochocki** [comment]
> so, it is a good idea to add similar accuracy improvements to root numbers?
> 
> Sqrt (2)=(visualized sqrt)(2)
> sqrt (2.5) = sqrt (10/4) = (visualized sqrt)(10)/2
> sqrt (3)/sqrt (2) = (visualized sqrt)(6)/2

@lzl1918 I think this would be good to include in the discussion for #170

**Comment by grochocki** [comment]
We believe this is dependent on #170, which in turn, is likely dependent on the ability to display rich math symbols and expressions in the output. We can move forward thinking through the UX and behavior assuming those issues are resolved. A couple additional pieces of feedback:
* We should consider supporting this across more modes (i.e., should not limit to just scientific mode).
* While a decimal like 0.123456789 _could_ be represented as 123456789/1000000000, it might not be the best user experience, so we should think through how we handle those cases.

**Comment by SaprativRay** [comment]
> Thanks for the clarification @whutddk. It seems this #170 is focused on making fractions/exponents visually distinct, while your idea covers covnersion between fractions and decimals. Let's keep this idea open for discussion so the community has the chance to provide feedback.
> 
> As @Luuk34 points out, most decimals do not work out to a nice conversion. While technically correct, I am not sure displaying this is the best user expereince:
> 
> > 1000048602673147023086269744836 / 10000000000000000000000000000000

While this is true, most CASIO calculators simply don't convert the decimals which do not have accurate and "aesthetic" representation in fractions. On pressing the button to convert, it doesn't do anything. While that might not be the best user experience, and I know this would be a lot of work for a minor feature, it makes sense to not convert but simply add a small lightbulb icon beside the same decimal value in the answer box which explains why the conversion was not performed.

**Comment by Rahul-Dixit** [comment]
Here is an example from the Google Calculator for Android:

![Screenshot_2020-08-14-15-40-56-700_com.google.android.calculator.jpg](https://user-images.githubusercontent.com/55381546/90493195-ae432480-e15f-11ea-8170-7bb32f7aec48.jpg)

Note that there is no button for switching decimal/fractional modes. It shows both forms by default.

**Comment by Chips1234** [comment]
I'm thinking something where the default is in decimal form, but there's a little button (perhaps with the symbol a and b over c, see image below of my Sharp Calculator's button) and when that's pressed, the decimal will turn into a fraction. Perhaps the default behaviour for the button is a mixed fraction, but when the [2nd] button is clicked, the button will turn mixed into improper fractions?
![image](https://user-images.githubusercontent.com/47545798/127580413-59d0e440-466a-4162-9af6-1f93dcdd69d7.png)
